cpp/lib/src/Zotero.cc

```diff
@@ -477,6 +477,16 @@ std::pair<unsigned, unsigned> MarcFormatHandler::processRecord(const std::shared
         } else if (key_and_node.first == "journalAbbreviation") {
             new_record.insertField("773", { { 'p', JSON::JSONNode::CastToStringNodeOrDie(key_and_node.first,
                                                                                          key_and_node.second)->getValue() } });
+        } else if (key_and_node.first == "extra") {
+            // comment field, can contain anything, even DOI's (e.g. for a webpage which has no DOI field)
+            const std::string extra(JSON::JSONNode::CastToStringNodeOrDie(key_and_node.first,
+                                                                          key_and_node.second)->getValue());
+            static RegexMatcher * const doi_matcher(RegexMatcher::RegexMatcherFactory("^DOI:\\s*([0-9a-zA-Z./]+)$"));
+            if (doi_matcher->matched(extra)) {
+                CreateSubfieldFromStringNode(key_and_node, "856", 'u', &new_record);
```

Comment:
check BSZ definition before merging this!
- loc.gov: 856u => https://www.loc.gov/marc/holdings/hd856.html
- bsz: 024 $2doi => https://wiki.bsz-bw.de/lib/exe/fetch.php?media=v-team:daten:datendienste:pica_marc21_rda_titel.xls